cypress/e2e/bookStore.cy.js

```diff
@@ -1,11 +1,57 @@
 /// <reference types='cypress' />
 
 describe('Book Store app', () => {
-  before(() => {
+  const user = {
+    username: 'Dmytro',
+    password: 'WordP@ss'
+  };
 
+  const book = {
+    title: 'Speaking JavaScript',
+    author: 'Axel Rauschmayer',
+    publisher: '0\'Reilly Media',
+    description: 'Like it or not, JavaScript is everywhere these days'
+  };
+
+  const alertMessage = {
+    added: 'Book added to your collection.',
+    deleted: 'Book deleted.'
+  };
+
+  beforeEach(() => {
+    cy.visit('/login');
   });
 
-  it('', () => {
+  it('should let user to login', () => {
+    cy.findByPlaceholder('UserName').type(user.username);
+    cy.findByPlaceholder('Password').type(user.password);
+    cy.get('#login').click();
+    cy.get('#userName-value').should('contain', user.username);
+    cy.url().should('include', '/profile');
+  });
+
+  it('should add a book to the collection', () => {
+    cy.viewport(750, 600);
+    cy.login();
+    cy.visit('/profile');
+    cy.contains('#item-2', 'Book Store').click();
+    cy.findByPlaceholder('Type to search').type(book.title);
+    cy.contains('a', book.title).click();
+    cy.get('#title-wrapper').should('contain', book.title);
+    cy.get('.btn.btn-primary').eq(2).click();
+    cy.on('window:alert', (alert) => {
+      expect(alert).to.equal(alertMessage.added);
+    });
+  });
 
+  it('should delete a book from the collection', () => {
+    cy.login();
```

Comment:
The login function is called without parameters. This function might require user credentials as parameters.

Reply:
Thank you! Fixed.